### DISCUSSION TITLE
Unable to run built jar

### DISCUSSION THREAD

**Discussion by SerenModz21** [FabricMC/fabric-api#2378] [Mod Dev Support]
hello, i've recently been making a server mod using fabric and all goes well during development (using the runServer task), however, when i run the build task and test the jar on a fabric server, i receive this error which seems to occur on all dependencies that are specified for minecraft:
<https://sourceb.in/Du7S8Up4XA>

if u require my build.gradle, it is originally from the mod example:
<https://sourceb.in/RAZRvIaSAk>

**Comment by warjort** [comment]
````Caused by: java.lang.ClassNotFoundException: de.maxhenkel.configbuilder.ConfigBuilder````

I guess you need to install this mod on your server as well?

````implementation "de.maxhenkel.configbuilder:configbuilder:${project.configbuilder_version}"````

Or if it is just a java library, you could use something like shadow to have everything in one jar.
https://imperceptiblethoughts.com/shadow/introduction/

**Reply by SerenModz21** [reply]
how would i get shadow to work with fabric loom?

**Reply by warjort** [reply]
Have you tried putting the configbuilder jar in your mods folder to test it solves your problem?

I mentioned shadow because it could be one option to make your mod easier to install.
It takes multiple jars and merges them into one.
I don't use it myself. I just know it exists. You would have the read the docs for which magic incantations to use. :-)

**Reply by SerenModz21** [reply]
i seemed to have gotten shadow to work, with my new build.gradle being https://sourceb.in/yqxrngdm4R. thank you so much for your help!
